### ISSUE TITLE
Oracle: inline PRIMARY KEY/UNIQUE with USING INDEX is not parsed in column definitions

### ISSUE THREAD

**Issue by saulotoledo** [sqlfluff/sqlfluff#8425]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQLFluff correctly supports USING INDEX for table-level constraints, but it fails for Oracle column-level inline constraints:

```sql
CREATE TABLE t_demo (
    id NUMBER
        CONSTRAINT pk_t_demo PRIMARY KEY
        USING INDEX TABLESPACE idx_ts
);
```

The same syntax appears valid in Oracle and should parse without error. Table-level forms already parse successfully, so this is a missing Oracle dialect rule for inline column constraints.

### Expected Behaviour

SQLFluff should parse Oracle inline column constraints with USING INDEX without producing an unparsable section.

### Observed Behaviour

The parser rejects the column-level form and leaves part of the statement as unparsable, even though the equivalent table-level form is accepted.

### How to reproduce

1. Save the following as repro.sql:

```sql
CREATE TABLE t_demo (
    id NUMBER
        CONSTRAINT pk_t_demo PRIMARY KEY
        USING INDEX TABLESPACE idx_ts,
    code VARCHAR2(10)
);
```

2. Run:

```sh
sqlfluff parse repro.sql --dialect oracle
```

### Dialect

oracle

### Version

4.3.0

### Configuration

Default.

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
